src/charm.py

```diff
@@ -43,6 +44,77 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.reconciliation)
         self.framework.observe(self.on.spring_boot_app_pebble_ready, self.reconciliation)
+        self.database: typing.Optional[DatabaseRequires] = self._setup_database_requirer(
+            "mysql", "spring-boot"
+        )
+
+    def _setup_database_requirer(self, relation_name: str, database_name: str) -> DatabaseRequires:
+        """Set up a DatabaseRequires instance.
+
+        The DatabaseRequires instance is an interface between the charm and various data providers.
+        It handles those relations and emit events to help us abstract these integrations.
+
+        Args:
+            relation_name: Name of the data relation
+            database_name: Name of the database (can be overwritten by the provider)
+
+        Returns:
+            DatabaseRequires object
+        """
+        database_requirer = DatabaseRequires(
+            self,
+            relation_name=relation_name,
+            database_name=database_name,
+        )
+        self.framework.observe(database_requirer.on.database_created, self._on_database_created)
+        self.framework.observe(self.on[relation_name].relation_broken, self._on_relation_broken)
+        return database_requirer
+
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Event triggered when a database was created for this application.
+
+        Args:
+            event: The DatabaseCreatedEvent object
+        """
+        logger.debug("Database credentials are received: %s", event.username)
+        self.reconciliation(event)
+
+    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
+        """Handle relation broken event.
+
+        Args:
+            event: The RelationBrokenEvent object
+        """
+        logger.debug("Relation with database removed: %s", event.relation.name)
+        self.reconciliation(event)
+
+    def _datasource(self) -> dict[str, str]:
+        """Compute datasource dict and return it.
+
+        Returns:
+            Dict containing details about the data provider integration
+        """
+        if self.database:
+            relations_data = list(self.database.fetch_relation_data().values())
+            if relations_data:
+                # There can be only one database integrated at a time
+                # cf: metadata.yaml
```

Comment:
It's a reminder that metadata.yml enforces only one mysql integration at a time

Reply:
I asked Niels to add a comment as one would wonder why we only take the first. We take the first because there's only 1 allowed integration at a time for this integration. And we trust juju to enforce this

Reply:
What does `cf` stand for? That is the main thing I was confused by

Reply:
It's [an abbreviation](https://en.wikipedia.org/wiki/Cf.) to give a link to reference material

Reply:
For those of us not familiar with latin 😉, would `see` also work?

Reply:
Yeap I think `cf` is mostly used in french